### PR TITLE
[BUG]: fix major bugs in PyKE

### PR DESCRIPTION
**Pull request by mirca**
cc @barentsen @mrtommyb

### CODE REVIEW

File: kepregr.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 from pyraf import iraf
 import numpy, sys, time, pyfits, pylab
-from pyfits import *
+from astropy.io import fits as pygfits
```
Comment (barentsen) [inline kepregr.py]: do you mean `pyfits` instead of `pygfits`?

**Comment by mirca** [comment]
The last commit addresses the changes on the ``Header.update`` method. http://docs.astropy.org/en/stable/io/fits/api/headers.html#astropy.io.fits.Header.update

**Comment by barentsen** [comment]
Thanks @mirca!  All these changes look sensible to me!

**Comment by mirca** [comment]
I just repeated [this](https://keplerscience.arc.nasa.gov/PyKEprimerWalkthroughA.shtml) example and right now the code is working smoothly :)
kepcotrend

**Comment by brunofisico** [comment]
Opa! 
Td finalizado? 
Obg Zé. Vou atualizar pelo seu GitHub 
Abraço 

Enviado do meu iPhone

**Comment by mirca** [comment]
Oi Bruno,

tudo bem?

Estou trabalhando no PyKE, mas seria muito bom ter um feedback externo ;-)

Abraço

**Comment by brunofisico** [comment]
Até amanhã eu te dou um retorno!
Cheers


Enviado do meu iPhone

> Em 11 de abr de 2017, às 19:33, Zé Vinícius <notifications@github.com> escreveu:
> 
> Oi Bruno,
> 
> tudo bem?
> 
> Estou trabalhando no PyKE, mas seria muito bom ter um feedback externo ;-)
> 
> Abraço
> 
>

**Comment by brunofisico** [comment]
Good stuff. So far testing the targets in kepdynamics has been all right and very good. Keep going! Congratulations!

**Comment by mirca** [comment]
```
KEPTRANSIT -- inputfile=kepflatten.fits outputfile=keptransit.fits datacol=DETSAP_FLUX errorcol=DETSAP_FLUX_ERR periodini_d=2.47 rprsini=0.1 T0ini=631.7 Eccini=0.0 arsini=7.0 incini=90.0 omegaini=0.0 LDparams=0.4330 -0.1022 secini=6.5e-06 fixperiod=no fixrprs=no fixT0=no fixEcc=no fixars=no fixinc=no fixomega=no fixsec=no fixfluxoffset=yes removeflaggeddata=yes ftol=0.0001 fitter=fmin norm=no plot=y clobber=y logfile=keptransit.log

KEPTRANSIT started at: Wed Apr 19 10:36:08 2017
cleaning done: about to fit transit
Optimization terminated successfully.
         Current function value: 9904.358989
         Iterations: 527
         Function evaluations: 845
Fit period (days) = 2.47061663998
Fit R_planet / R_star = 0.111345870865
Fit T0 (BJD) = 631.71288525
Fit eccentricity = -0.00420566821188
Fit a / R_star = 14.7154742705
Fit inclination (deg) = 89.5397116211
Fit omega = -0.000888779631521
Fit seconary eclipse depth = -3.75392563322e-09
KEPTRANSIT completed at: Wed Apr 19 10:37:33 2017
```

figure_1

**Comment by mirca** [comment]
@barentsen Are you happy if I merge this as is or would you like to play with it before? x)

File: keptrial.py [hunk 2]
```diff
@@ -141,48 +118,40 @@ def keptrial(infile,outfile,datacol,errcol,fmin,fmax,nfreq,method,
...
-            clf()
-	    pylab.axes([0.08,0.08,0.88,0.89])
-            pylab.gca().xaxis.set_major_formatter(pylab.ScalarFormatter(useOffset=False))
-            pylab.gca().yaxis.set_major_formatter(pylab.ScalarFormatter(useOffset=False))
-            n,bins,patches = pylab.hist(freq,bins=nfreq,range=[fmin,fmax],
-                                        align='mid',rwidth=1,ec='#0000ff',
-                                        fc='#ffff00',lw=2)
+            plt.ion()
+	    plt.figure(1,figsize=[7,10])
+            plt.clf()
```
Comment (barentsen) [inline keptrial.py]: The indentation is a bit funny here?

File: kepbls.py [hunk 3]
```diff
@@ -250,60 +251,56 @@ def kepbls(infile,outfile,datacol,errcol,minper,maxper,mindur,maxdur,nsearch,
...
 
     if status == 0 and plot:
-        pylab.plot(ptime[1:-1],pout[1:-1],color=lcolor,linestyle='-',linewidth=lwidth)
-        pylab.fill(ptime,pout,color=fcolor,linewidth=0.0,alpha=falpha)
-	pylab.xlabel(xlab, {'color' : 'k'})
-        pylab.ylabel(ylab, {'color' : 'k'})
-        pylab.grid()
+        plt.plot(ptime[1:-1],pout[1:-1],color=lcolor,linestyle='-',linewidth=lwidth)
+        plt.fill(ptime,pout,color=fcolor,linewidth=0.0,alpha=falpha)
+	plt.xlabel(xlab, {'color' : 'k'})
```
Comment (mirca) [inline kepbls.py]: bang

**Comment by mirca** [comment]
@barentsen @mrtommyb merging in 3, 2, 1...

**Pull request merged**